packages/linter/src/rule.ts

```diff
@@ -0,0 +1,46 @@
+import { Context, Result, Status } from "./index";
+
+// Don't completely understand why we need to do this instead of "typeof Rule"
+// (should be equivalent??) but whatever. See
+// https://stackoverflow.com/a/52358194/11543
+export interface RuleConstructor {
+  new (): Rule;
+}
+
+export abstract class Rule {
+  static href = "";
+  static info = "";
+  abstract run(context: Context): Promise<Result | Array<Result>>;
+  protected pass(s?: string) {
```

Comment:
Would marking `pass` as `async` have the same  effect as returning a promise?

Reply:
Yes, it does! Change in 3f54e73.